src/EnergyPlus/OutputReportTabular.cc

```diff
@@ -17269,47 +17258,45 @@ std::string DateToString(int const codedDate) // word containing encoded month,
     int Minute; // minute in integer format (0:59)
     std::string monthName;
 
-    if (codedDate != 0) {
-        DecodeMonDayHrMin(codedDate, Month, Day, Hour, Minute);
-        --Hour;
-        if (Minute == 60) {
-            ++Hour;
-            Minute = 0;
-        }
-        if (Month == 1) {
-            monthName = "JAN";
-        } else if (Month == 2) {
-            monthName = "FEB";
-        } else if (Month == 3) {
-            monthName = "MAR";
-        } else if (Month == 4) {
-            monthName = "APR";
-        } else if (Month == 5) {
-            monthName = "MAY";
-        } else if (Month == 6) {
-            monthName = "JUN";
-        } else if (Month == 7) {
-            monthName = "JUL";
-        } else if (Month == 8) {
-            monthName = "AUG";
-        } else if (Month == 9) {
-            monthName = "SEP";
-        } else if (Month == 10) {
-            monthName = "OCT";
-        } else if (Month == 11) {
-            monthName = "NOV";
-        } else if (Month == 12) {
-            monthName = "DEC";
-        } else {
-            monthName = "***";
-        }
-        StringOut = format("{:02}-{:3}-{:02}:{:02}", Day, monthName, Hour, Minute);
-        if (has(StringOut, "*")) {
-            StringOut = "-";
-        }
-    } else { // codeddate = 0
-        StringOut = "-";
-    }
+    if (codedDate == 0) {
+        return "-";
+    }
+
+    DecodeMonDayHrMin(codedDate, Month, Day, Hour, Minute);
+    --Hour;
+    if (Minute == 60) {
+        ++Hour;
+        Minute = 0;
+    }
+    switch (Month) {
```

Comment:
Ay Caramba! `monthStrings[Month-1]` anyone?

Reply:
This was a quick conversation from existing code, but I can add a `static constexpr std::array<std::string_view, 12> Months{"JAN", "FEB", "MAR", "APR", "MAY", "JUN", "JUL", "AUG", "SEP", "OCT", "NOV", "DEC"};` in and use that.